OSF download

billbrod opened the pull request:
Expands section about data, including more details about how to use OSF for hosting external data, with example functions for downloading and extracting them.

billbrod committed:
adds more detils about data

billbrod committed:
adds warning

BalzaniEdoardo commented:
That's awesome! I will use this soon hopefully. 
consider creating an example in the gallery that downloads a  relatively small dataset on OSF (a few MBs), and reference that one as a tutorial. 
An idea could be moving your description of the python/command line commands to download the data to the example, and reference the example in the note, so that people can actively run the python version of the data pipeline.
Let me know if you like the suggestion!

BalzaniEdoardo reviewed (changes requested):
I'll repeat my comment in the conversation:

That's awesome! I will use this soon hopefully.
consider creating an example in the gallery that downloads a relatively small dataset on OSF (a few MBs), and reference that one as a tutorial.
An idea could be moving your description of the python/command line commands to download the data to the example, and reference the example in the note, so that people can actively run the python version of the data pipeline.
Let me know if you like the suggestion!

billbrod committed:
adds plot_osf example

billbrod committed:
adds pillow dependency for docs

billbrod commented:
@BalzaniEdoardo I think that's a good idea, went ahead and added an example. Let me know what you think! Two notes: 
- I named it `plot_osf.py` because of the issue you mentioned in #37, where, by default, only files starting with "plot" are run. It's not actually plotting anything, so it's confusing here. Where does the configuration live?
- I'm not sure how much belongs in the note now, as opposed to the example. Double-check how I split them up and let me know what you think.

billbrod committed:
Merge branch 'main' into osf_download

BalzaniEdoardo commented:
> @BalzaniEdoardo I think that's a good idea, went ahead and added an example. Let me know what you think! Two notes:
> 
>     * I named it `plot_osf.py` because of the issue you mentioned in [mkdocs-gallery sort tutorials & configs #37](https://github.com/flatironinstitute/ccn-template/issues/37), where, by default, only files starting with "plot" are run. It's not actually plotting anything, so it's confusing here. Where does the configuration live?
> 
>     * I'm not sure how much belongs in the note now, as opposed to the example. Double-check how I split them up and let me know what you think.

For the configuration, that will be my second-to-next contribution to the template. If I have to bet is the config lives  [here](https://github.com/smarie/mkdocs-gallery/blob/main/src/mkdocs_gallery/gen_gallery.py), and you have to set the dictionary key "filename_pattern" in the [conf script](https://github.com/smarie/mkdocs-gallery/blob/main/docs/gallery_conf.py):
```python
import os
import sys

import plotly.io as pio
pio.renderers.default = 'sphinx_gallery'

from mkdocs_gallery.gen_gallery import DefaultResetArgv

min_reported_time = 0
if 'SOURCE_DATE_EPOCH' in os.environ:
    min_reported_time = sys.maxint if sys.version_info[0] == 2 else sys.maxsize

# To be used as the "base" config,
# this script is referenced in the mkdocs.yaml under `conf_script` option: docs/gallery_conf.py
conf = {
    'reset_argv': DefaultResetArgv(),
    'min_reported_time': min_reported_time,
    'filename_pattern': re.escape(os.sep) + "YOUR_PATTERN",
}
```

BalzaniEdoardo commented on docs/notes/07-data.md:
```diff
@@ -1,62 +1,48 @@
 # Data
 
```
I would Add a section title
```suggestion

## Data Management Practices

```

BalzaniEdoardo commented on docs/examples/plot_osf.py:
```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/env python3
+"""Downloading data from the OSF
+=============================
+
```
Generally, the way you split the information makes a lot of sense. In my opinion a few lines of intro on OSF and then pointing to the note would be beneficial even here. It is information repetition, but it would make this tutorial almost stand-alone

BalzaniEdoardo reviewed (changes requested):
great example, it is going to be very useful for my package as well. I left a couple of notes, but it is almost good to go.

billbrod commented:
Coming back to this much later: instead of using custom functions to download from OSF, use [pooch](https://www.fatiando.org/pooch/latest/index.html). I'm changing plenoptic to use pooch as part of [this PR](https://github.com/LabForComputationalVision/plenoptic/pull/252) and it's very effective.
